solr/core/src/java/org/apache/solr/core/ZkContainer.java

```diff
@@ -279,6 +287,39 @@ public void close() {
     }
     
   }
+  public SolrMetricProducer getMetricsProducer() {
+    return this.metricProducer;
+  }
+
+  public static final class ZkClientMetrics implements SolrInfoBean, SolrMetricProducer {
```

Comment:
Is there a benefit to having this class exposed as a SolrInfoBean? Could maybe remove this intermediate class entirely (see above)?

Reply:
true.